### DISCUSSION TITLE
Developer Tools needed

### DISCUSSION THREAD

**Discussion by ArnieGA** [mobxjs/mobx#3126] [Ideas]
Hi:

Mobx is awesome!...until the part where you lack good dev tools for your observables.  Andy Kogut's extension appears to be the sole option out there, which hasn't been successful coping with React 17 changes...

Could you please add this to your future milestones list?

I'm willing to bet that I'm not the only developer who thinks Mobx is great, but lacks the proper browser development tools. I go crazy every time I have to code a complex React component observer, because I know that I will not have the tool to keep an eye on the corresponding observers.

**Comment by codeBelt** [comment]
That would cool! I saw this one a while back:

https://github.com/zalmoxisus/mobx-remotedev

**Reply by ArnieGA** [reply]
@codeBelt Thanks for the tip!  Cheers!...
However, it feels like Mobx as a solution to client state/store should include this out of the box.  It's incomplete without it.  Additionally, your suggestion requires additional configuration via JS which should be unnecessary for obvious reasons (although acceptable if strictly necessary for a certain project/situation).

I don't know... could someone please second this if agreeable so that the devs possibly pay at least some attention?
Thanks M8's!! :-)

**Reply by kubk** [reply]
@codeBelt Unfortunately this extension is not maintained anymore. Also it doesn't show computeds at all.

**Comment by kubk** [comment]
@ArnieGA Completely agree with you. Having devtools would be nice. 
Andy Kogut's extension displays to much confusing information, here is what happens for me when I call a simple action with 1 computed:

![image](https://user-images.githubusercontent.com/22447849/135705101-e0b41010-92e5-47c4-98ef-03735fbf6558.png)

It's not clear what's going on. I decided to create a simple logger to understand what is going on with application state. It's is still work in progress but you can try: https://github.com/kubk/mobx-log

Here is the log output:

![image](https://user-images.githubusercontent.com/22447849/135744086-a9ec9426-5358-4dad-9158-cbbc3eb21403.png)

Here we see that an action has modified observable, it triggered computed to recalculate.

**Reply by ArnieGA** [reply]
Let me tell ya, if it weren't for the fact that the console gets too crowded, this would be THE perfect solution.  I've installed this and I'm currently using it; so far so good:  minimal setup/config, color coded, clear and concise results, just brilliant!.  But honestly, this should be refactored into a browser extension so that it can live in its own universe instead of the "public plaza" ;-) The console is getting too crowded very very fast.

**Reply by kubk** [reply]
@ArnieGA you can use built-in filter functionality to filter by some specific store:

![image](https://user-images.githubusercontent.com/22447849/135743439-8534ab2a-81ef-4a07-83e2-0a41ba722faf.png)

But the whole idea of the logger is that you put `makeLoggable` only for needed classes and remove it after resolving bug.

Regarding browser devtools. I completely agree with you. But it is very complex utility and takes time to develop. Unfortunately it looks like there no people who are willing to spend so much time for free at the moment :(

**Comment by TechAkayy** [comment]
I come from the vue.js world where the community can't live without the vue-devtools which is like the soul of DX. Official integration into redux-devtools with remotedev (computed not showing currently and not stable) is very important, esp when mobx is sooo good without any other good alternative...
